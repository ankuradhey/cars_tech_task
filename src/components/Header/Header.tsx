import React, { FC } from "react";
import { Link } from "react-router-dom";
import logo from "../../assets/logo.png";

export const Header: FC = () => {
    return (
        <div className="d-flex flex-column flex-md-row align-items-center p-3 px-md-4 mb-3 bg-white border-bottom box-shadow">
            <Link className="brand my-0 mr-md-auto font-weight-normal" to="/">
                <img src={logo} alt="Auto1.com" />
            </Link>
            <nav className="my-2 my-md-0 mr-md-3">
                <Link className="p-2 text-dark" to="/">
                    Purchase
                </Link>
                <Link className="p-2 text-dark" to="/">
                    My Orders
                </Link>
                <Link className="p-2 text-dark" to="/">
                    Sell
                </Link>
            </nav>
        </div>
    );
};
