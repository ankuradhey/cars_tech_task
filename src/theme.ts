import { DefaultTheme } from "styled-components";

export const defaultTheme: DefaultTheme = {
    borderRadius: "2px",
    palette: {
        common: {
            black: "#4A4A4A",
            white: "#EDEDED",
        },
        primary: {
            main: "#EA7F28",
            contrastText: "#ffffff",
        },
        secondary: {
            main: "#D37324",
            contrastText: "#ffffff",
        },
        tertiary: {
            main: "#ffffff",
            contrastText: "#EDEDED",
        },
    },
    gutterSize: {
        sm: "8px",
        md: "12px",
        lg: "24px",
    },
};
