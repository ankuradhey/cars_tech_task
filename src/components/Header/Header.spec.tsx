import React from "react";
import { Router } from "react-router-dom";
import { createMemoryHistory } from "history";
import { render } from "@testing-library/react";
import { ThemeProvider } from "styled-components";
import { defaultTheme } from "../../theme";
import { Header } from "./Header";

describe("Header Component", () => {
    it("Should render Header component", () => {
        const history = createMemoryHistory();
        const route = "/";
        history.push(route);

        const { container, getByText } = render(
            <ThemeProvider theme={defaultTheme}>
                <Router history={history} initialEntries={["/"]}>
                    <Header />
                </Router>
            </ThemeProvider>
        );
        expect(getByText("Purchase")).toBeInTheDocument();
        expect(getByText("My Orders")).toBeInTheDocument();
        expect(getByText("Sell")).toBeInTheDocument();
        expect(container.firstChild).toMatchSnapshot();
    });
});
