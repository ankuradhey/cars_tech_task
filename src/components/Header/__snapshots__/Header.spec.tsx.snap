// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component Should render Header component 1`] = `
<div
  class="d-flex flex-column flex-md-row align-items-center p-3 px-md-4 mb-3 bg-white border-bottom box-shadow"
>
  <a
    class="brand my-0 mr-md-auto font-weight-normal"
    href="/"
  >
    <img
      alt="Auto1.com"
      src="logo.png"
    />
  </a>
  <nav
    class="my-2 my-md-0 mr-md-3"
  >
    <a
      class="p-2 text-dark"
      href="/"
    >
      Purchase
    </a>
    <a
      class="p-2 text-dark"
      href="/"
    >
      My Orders
    </a>
    <a
      class="p-2 text-dark"
      href="/"
    >
      Sell
    </a>
  </nav>
</div>
`;
